sisl/io/xsf.py

```diff
@@ -386,11 +475,58 @@ class axsfSile(xsfSile):
     """
 
     def _setup(self, *args, **kwargs):
-        # Correct number of steps
         super()._setup(*args, **kwargs)
-        if not hasattr(self, '_md_steps'):
+        if self._md_steps is None:
+            # The user didn't specify steps on instantiation.
+            # In read mode it doesn't really matter.
+            # In write mode it needs to be the correct number, so we should error out in this case.
+            # However sisl requires instantiation to be possible without additional args in its tests,
+            # so we need to assume that only one step is intended (as in the tests) in this case.
             self._md_steps = 1
```

Comment:
@zerothi I put an explanation in a code comment. The referred test is `test_read_write_geometry` which works over all siles that can read and write geometries.

Reply:
Yeah, I saw that you check for the variable `md_steps`, perhaps we should then define a naming convention for a specific "steps" parameter. I.e. in FC runs one still have multiple steps, but I wouldn't consider them `md_steps`.

Could we name it `_geometry_steps`/`_ngeometry`/`_geometry_count` for clarity? This should then be applied to all files capable of having multiple geometries.

Reply:
In this case axsf and xsf share all their code, so they check for whether the steps variable (whichever we call it) is None. Should this also be standardized then? Ie. that all siles capable of loading geometries must have `_geometry_count` set, either to `None` if they don't support it, an integer if they do support it but with a catch that it might be incorrect if the sile is in read-mode (because then that number hasn't been read)?

Reply:
hmmm. good, question, probably the cleanest way is to have `axsf` inherit the `xsf` and then decouple them entirely... What do you think?

It seems rather intrusive to have all siles have the `_geometry_count` variable if it ain't used...